controllers/Auth.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-param-reassign */
+import bcrypt from 'bcrypt';
+import nodemailer from 'nodemailer';
+import Response from '../helpers/Response';
+import codes from '../helpers/statusCode';
+import User from '../database/models/User';
+import TokenUtil from '../helpers/TokenUtil';
+
+const AuthController = {
+  signup: async (req, res) => {
+    const {
+      firstname, lastname, othername, gender, email,
+      password: pass, phonenumber, passport: passporturl,
+    } = req.body;
+
+    try {
+      const password = bcrypt.hashSync(pass, 10);
+      const user = await User.create({
+        firstname, lastname, othername, gender, email, password, phonenumber, passporturl,
+      });
+      delete user.password;
+      const token = TokenUtil.sign(user);
+      return Response.send(res, codes.created, {
+        data: { token, user },
+      });
+    } catch (error) { return Response.handleError(res, error); }
+  },
+
+  login: async (req, res) => {
+    const { email, password } = req.body;
+    try {
+      const user = await User.where(['email', '=', email]).first();
+
+      if (!user || !await bcrypt.compareSync(password, user.password)) {
+        return Response.send(res, codes.unAuthorized, {
+          error: 'Invalid email address or password.',
+        });
+      }
+      delete user.password;
+      const token = TokenUtil.sign(user);
+      res.cookie('token', token);
+      return Response.send(res, codes.success, {
+        data: { token, user },
+      });
+    } catch (error) { return Response.handleError(res, error); }
+  },
+
+  requestResetPassword: async (req, res) => {
+    const { email } = req.body;
+    try {
+      const user = await User.where(['email', '=', email]).first();
+
+      if (!user) {
+        return Response.send(res, codes.notFound, {
+          error: 'User does not exist.',
+        });
+      }
+
+      const emailtoken = Math.random().toString(15).substring(2);
+      const transporter = nodemailer.createTransport({
```

Comment:
You can abstract this into a separate file.

Reply:
Okay, I will do just that